src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -91,16 +91,18 @@ class StoreAppender extends AbstractCloseable
 
         try {
             queue.cleanupStoreFilesWithNoData();
+            // Avoid normalizing last existing segment which may be appended later on via queue replication
+            //  even if it is in the past
+            int lastCycle = queue.lastCycle();
+            cycle = Math.min(queue.cycle(), lastCycle >= 0 ? lastCycle : queue.cycle());
```

Comment:
should we try and use a name other than lastCycle for this variable, considering that's also the name of a field in the class?

Reply:
perhaps even put these two lines into a method with a name that describes what it's doing?

Reply:
It does seem suspicious and I will double recheck it.